src/ring_theory/integral_closure.lean

```diff
@@ -135,14 +140,19 @@ theorem is_integral_alg_equiv {A B : Type*} [ring A] [ring B] [algebra R A] [alg
 ⟨λ h, by simpa using is_integral_alg_hom f.symm.to_alg_hom h, is_integral_alg_hom f.to_alg_hom⟩
 
 theorem is_integral_of_is_scalar_tower [algebra A B] [is_scalar_tower R A B]
-  (x : B) (hx : is_integral R x) : is_integral A x :=
+  {x : B} (hx : is_integral R x) : is_integral A x :=
 let ⟨p, hp, hpx⟩ := hx in
 ⟨p.map $ algebra_map R A, hp.map _,
   by rw [← aeval_def, aeval_map_algebra_map, aeval_def, hpx]⟩
 
+lemma is_integral_int_alg_hom_class {C F : Type*} [comm_ring C] [algebra A B] [algebra A C] {b : B}
+  [alg_hom_class F A B C] (f : F) (hb : is_integral ℤ b) :
+  is_integral ℤ (f b) :=
+is_integral_alg_hom f hb
```

Comment:
Can you remove `[algebra]` and replace `[alg_hom_class F A B C]` by `[ring_hom_class F B C]` here? And the rings don't have to be commutative? And maybe rename to `map_is_integral_int` and the one above to `map_is_integral`?

Reply:
I've generalized the result. But I don't see `map_is_integral_int`...